### PR TITLE
Fix leakage of SDK installations

### PR DESCRIPTION
**Pull request by cocreature**
This fixes an issue where the integration tests leaked SDK
installations. Easily verified by `ls /tmp/extra-dir*` before and
after running the tests.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: libs-haskell/test-utils/DA/Test/Util.hs [hunk 1]
```diff
@@ -44,7 +44,12 @@ withTempFileResource :: (IO FilePath -> TestTree) -> TestTree
 withTempFileResource f = withResource newTempFile snd (f . fmap fst)
 
 withTempDirResource :: (IO FilePath -> TestTree) -> TestTree
-withTempDirResource f = withResource newTempDir snd (f . fmap fst)
+withTempDirResource f = withResource newTempDir delete (f . fmap fst)
+  -- The delete action provided by `newTempDir` calls `removeDirectoryRecursively`
+  -- and silently swallows errors. SDK installations are marked read-only
+  -- which means that they don’t end up being removed which is obviously
+  -- not what we intend.
```
Comment (hurryabit) [inline libs-haskell/test-utils/DA/Test/Util.hs:51]: Is there a particular reason why we install SDK marked read-only? I actually find that quite inconvenient when I need to garbage collect SDK installations every now and then. Shouldn't we rather change the installation process?

Comment (cocreature) [reply libs-haskell/test-utils/DA/Test/Util.hs:51]: Figuring out why something goes wrong for a user is already challenging enough. If we add to that people that accidentally modified their installation we end up in a very messy situation. Currently, it’s at least a very deliberate choice if you do that.

As for garbage collecting, I believe that is better solved by adding commands to the assistant for doing that and not by making it easier to delete things manually.

Comment (hurryabit) [reply libs-haskell/test-utils/DA/Test/Util.hs:51]: That makes sense. Thanks.

**Pull request merged**